base/traits.jl

```diff
@@ -2,29 +2,29 @@
 
 ## numeric/object traits
 # trait for objects that have an ordering
-abstract type TypeOrder end
-struct HasOrder <: TypeOrder end
-struct Unordered <: TypeOrder end
+abstract type OrderStyle end
+struct HasOrder <: OrderStyle end
+struct Unordered <: OrderStyle end
 
-TypeOrder(instance) = TypeOrder(typeof(instance))
-TypeOrder(::Type{<:Real}) = HasOrder()
-TypeOrder(::Type{<:Any}) = Unordered()
+OrderStyle(instance) = OrderStyle(typeof(instance))
+OrderStyle(::Type{<:Real}) = HasOrder()
+OrderStyle(::Type{<:Any}) = Unordered()
 
 # trait for objects that support arithmetic
-abstract type TypeArithmetic end
-struct ArithmeticRounds <: TypeArithmetic end     # least significant bits can be lost
-struct ArithmeticOverflows <: TypeArithmetic end  #  most significant bits can be lost
-struct ArithmeticUnknown <: TypeArithmetic end
+abstract type ArithmeticStyle end
+struct ArithmeticRounds <: ArithmeticStyle end     # least significant bits can be lost
+struct ArithmeticOverflows <: ArithmeticStyle end  #  most significant bits can be lost
```

Comment:
@StefanKarpinski also suggested renaming `ArithmeticOverflow` to `ArithmeticWraps` to distinguish it from saturating arithmetic in the future.